docs/source/troubleshooting/basic_chatops_troubleshooting.rst

```diff
@@ -0,0 +1,107 @@
+
+----------------------------------------------
+Troubleshooting using Hubot self-check script:
+----------------------------------------------
+
+We have a `self-check script <https://github.com/StackStorm/st2chatops/blob/master/scripts/self-check.sh>`_ 
+to help you debug chatops issue in StackStorm 1.5 and above.
+
+Just copy and run it on the server. It runs a few essential tests giving you basic troubleshooting steps in
+case of a failure.
+
+
+-------------------------------------------------
+Manual troubleshooting stepssame as  the script):
+-------------------------------------------------
+
+The assumption is that StackStorm client can connect to the server instance. Authentication is in place and
+``st2 run core.local cmd=echo`` succeeds. (This is checked in the script mentioned above.). Following are the
+issues that users usually face with chatOps:
+
+
+Your bot is not online:
+-----------------------
+
+You've installed StackStorm, but Hubot didn't come online in your chatOps client (Slack, Hipchat etc..)
+
+    *Possible reasons:*
+
+    - Incorrect adapter settings. Check ``/opt/stackstorm/chatops/st2chatops.env`` for adapter settings.
+      For example, for slack adapter, we need to uncomment following lines and update slack token:
+
+        .. code-block:: shell
+
+           # Slack settings (https://github.com/slackhq/hubot-slack):
+           # export HUBOT_ADAPTER=slack
+           # export HUBOT_SLACK_TOKEN=xoxb-CHANGE-ME-PLEASE
+
+      After changing the adapter settings don't forget to restart the service using this command:
+        
+        .. code-block:: shell
+
+           $ sudo service st2chatops restart
+
+    - Also, make sure the login credentials and ``ST2_HOSTNAME`` are correct.
+    - Check if Hubot-stacksotrm is installed:
+
+        .. code-block:: shell
+
+           $ cd /opt/stackstorm/chatops && npm list | grep hubot-stackstorm
+
+      In case the installation is outdated or corrupt, install hubot-stackstorm:
+      
+        .. code-block:: shell
+        
+           $ cd /opt/stackstorm/chatops && npm install hubot-stackstorm
+
+
+There's no StackStorm commands in `!help`:
+--------------------------------------------
+
+Hubot is online, but when you say ``!help``, only default commands are listed (no commands starting with ``st2``).
+
+    *Possible reasons:*
+
+    - Hubot can't connect to StackStorm API. Look in the Hubot logs for errors: ``/var/log/st2/st2chatops.log``
+    - Commands aren't registered. Try ``st2ctl reload --register-all``.
+    - It's also possible you changed the bot's name or you have not invited your bot to a channel in the chatOps
+      client. 
+
+
+StackStorm commands throw errors:
+---------------------------------
+
+Hubot is online, you can see ``st2`` commands, but saying something like ``!st2 list actions``
+either throws an error, or gives an acknowledgement message without result, or no response at all.
```

Comment:
Those are very different cases when debugging manually. This kind of grouping only makes sense in an automated script.

> throws an error

Should be debugged according to the error, could be both client-side and server-side.

> gives an acknowledgement message without result

Stream error in 9 cases out of 10.

> no response at all

Either incorrect syntax on the user's side (the command just isn't typed correctly; 9 cases out of 10), or hubot-stackstorm regression (this is serious, and should be caught in our test; expect community to go mad if it's really an error in hubot-stackstorm).